lib/project_types/script/layers/infrastructure/languages/rust_task_runner.rb

```diff
@@ -9,15 +9,20 @@ class RustTaskRunner
           BUILD_TARGET = "wasm32-unknown-unknown"
           METADATA_FILE = "build/metadata.json"
           CARGO_BUILD_CMD = "cargo build --target=#{BUILD_TARGET} --release"
+          MIN_CARGO_VERSION = "1.50"
 
           def initialize(ctx, _)
             @ctx = ctx
           end
 
-          def dependencies_installed?
+          def project_dependencies_installed?
             true
           end
 
+          def check_system_dependencies!
+            check_tool_version!("cargo", MIN_CARGO_VERSION)
+          end
+
```

Comment:
We should add a test for this in [rust_task_runner_test.rb](https://github.com/Shopify/shopify-cli/blob/main/test/project_types/script/layers/infrastructure/languages/rust_task_runner_test.rb)